src/API.jl

```diff
@@ -917,9 +925,21 @@ function precompile(ctx::Context)
             break
         end
         if stale
-            Base.compilecache(pkg, sourcepath)
+            t = @async begin
```

Comment:
You should probably limit the parallelization here somewhat.  I suggest limiting to `Sys.CPU_THREADS + 1` by default, and limit with an environment variable or something.  `num_tasks = parse(Int, get(ENV, "JULIA_NUM_PRECOMPILE_TASKS", string(Sys.CPU_THREADS + 1)))` perhaps?

Reply:
Sounds good